### PR TITLE
[EventDispatcher] Fix BC/FC layer

### PR DESCRIPTION
**Pull request by chalasr**
| Q             | A
| ------------- | ---
| Branch?       | master
| Bug fix?      | yes
| New feature?  | no
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | n/a
| License       | MIT
| Doc PR        | n/a

Fixes
> TypeError: Argument 3 passed to Symfony\Component\EventDispatcher\EventDispatcher::doDispatch() must be an instance of Symfony\Component\EventDispatcher\Event

Spotted in https://github.com/lexik/LexikJWTAuthenticationBundle/pull/637

### CODE REVIEW

File: src/Symfony/Component/EventDispatcher/EventDispatcher.php [hunk 1]
```diff
@@ -229,14 +229,6 @@ public function removeSubscriber(EventSubscriberInterface $subscriber)
...
-    {
-        $this->doDispatch($listeners, $eventName, $event);
-    }
-
-    /**
-     * @deprecated since Symfony 4.3, use callListeners() instead
-     */
-    protected function doDispatch($listeners, $eventName, Event $event)
     {
         $stoppable = $event instanceof Event || $event instanceof ContractsEvent || $event instanceof StoppableEventInterface;
```
Comment (chalasr) [inline src/Symfony/Component/EventDispatcher/EventDispatcher.php]: only the first check can be true right now due to the `Event` typehint on `$event`

**Comment by nicolas-grekas** [comment]
This means doDidpatch() won't be called if we merge this patch, leading to a BC break.

**Comment by chalasr** [comment]
Indeed. The only way to keep calling `doDispatch()` in every case would be to make the deprecated `Event` class extend the one from contracts, which seems impossible due to the signature incompatibility; any suggestion welcome. 
Also is it expected that all core events still extend the deprecated class?

**Comment by chalasr** [comment]
Patch updated to keep calling `doDispatch()` in case `$event instanceof (deprecated) Event`. Not sure it's enough though

**Comment by nicolas-grekas** [comment]
We could check using reflection in the constructor if doDispatch exists while callListeners doesn't?

**Review by nicolas-grekas — APPROVED** [review]

**Comment by fabpot** [comment]
Thank you @chalasr.

**Comment by fabpot** [comment]
Thank you @chalasr.

**Pull request merged**